includes/discount-functions.php

```diff
@@ -32,19 +32,29 @@ function edd_get_discounts( $args = array() ) {
 
 	$args = wp_parse_args( $args, $defaults );
 
-	$discounts = get_posts( $args );
+	$discounts_hash = md5( json_encode( $args ) );
+	if ( ! $discounts = wp_cache_get( $discounts_hash, 'edd-discounts' ) ) {
```

Comment:
In order to maintain compatibility with PHP 5.3, please declare and call the function outside the `if` statement and then reference the variable.

i.e.
```php
$discounts = wp_cache_get( $discounts_hash, 'edd-discounts' )
if ( false === $discounts ) {
```